packages/clay-badge/src/ClayBadge.soy

```diff
@@ -0,0 +1,110 @@
+{namespace ClayBadge}
+
+/**
+ * This renders a badge.
+ */
+{template .render}
+	{@param? icon: [
+		alignment: string,
+		monospaced: bool,
+		spritemap: string,
+		symbol: string
+	]}
+	{@param? id: string}
+	{@param? label: html|string}
+	{@param? href: string}
+	{@param? shape: string}
+	{@param? size: string}
+	{@param? style: string}
+
+	{let $attributes kind="attributes"}
+		class="badge
+			{sp}badge-{$style ?: 'default'}
+
+			{if $shape == 'pill'}
+				{sp}badge-pill
+			{/if}
+
+			{if $size}
+				{sp}badge-{$size}
+			{/if}
+		"
+
+		{if $id}
+			id="{$id}"
+		{/if}
+
+		{if $href}
+			href="{$href}"
+		{/if}
+	{/let}
+
+	{if $href}
+		<a {$attributes}>
+			{call .content}
+				{param icon: $icon /}
+				{param label: $label /}
+			{/call}
+		</a>
+	{else}
+		<span {$attributes}>
+			{call .content}
+				{param icon: $icon /}
+				{param label: $label /}
+			{/call}
+		</span>
+	{/if}
+
+{/template}
+
+/**
+ * Renders icon content
+ */
+{template .content}
+	{@param? icon: [
+		alignment: string,
+		monospaced: bool,
+		spritemap: string,
+		symbol: string
+	]}
+	{@param? label: html|string}
+
+	{let $iconAlignment: $icon and $icon.alignment ? $icon.alignment : 'right' /}
+
+	{if $icon and $iconAlignment == 'left'}
+		<button aria-label="close" class="close" type="button">
+			{call .icon}
+				{param icon: $icon /}
+			{/call}
+		</button>
+	{/if}
+
+	{$label ? $label : ''}
+
+	{if $icon and $iconAlignment == 'right'}
+		<button aria-label="close" class="close" type="button">
```

Comment:
Hey @carloslancha,

ClayButton is not reusable. I would not be able to customize the `class`.